app/src/main/kotlin/com/wire/android/ui/WireActivity.kt

```diff
@@ -83,16 +83,18 @@ class WireActivity : AppCompatActivity() {
             navigationManager.navigateState
                 .onEach { command ->
                     if (command == null) return@onEach
-                    keyboardController?.hide()
                     navigateToItem(navController, command)
                 }
                 .launchIn(scope)
 
             navigationManager.navigateBack
                 .onEach {
-                    keyboardController?.hide()
-                    navController.popWithArguments(it)
+                    if (!navController.popWithArguments(it)) finish()
```

Comment:
A question, do we have anything that might be affected by finishing the activity by `finish()`?

Reply:
@vitorhugods do you have any comment?

Reply:
Killing the activity shouldn't do much IMHO. Same as the user killing the app by swiping it up.

If there are new events, a push will arrive, sync will be performed and we're back online with notifications being displayed.